### PR TITLE
[docs] added notes about params usage when data is provided via path and removed unused param

### LINKED ISSUE #2023: weight parameter not working
Good day,

I am interesting to use the weight param to set my weight for my objective function. I believe this weight give me the option to set individual weight to each row and this will in turn leads to higher weight == higher contribution in the terms of loss. 

Based on the documentation: 
image

I passed this as a dictionary params: (just a simple example). My weight column is the first column. 

```python 
params = {
        'nthread' : 6,
        'learning_rate': 0.01,
        'num_leaves':80,
        'max_depth': 7,
        'weight': 0}

```

and later used it in train. 

After training, when i check the feature importance and find that the column weight is used as a feature. Does this mean that the weight parameter is not taken into account? It will be good if you can help 

 Edit: is this weight_column suppose to be in .dataset and not in param dictionary?

### PR DESCRIPTION
**Pull request by StrikerRUS**
Closed #2023.

Usage example:

```
train = lgb.Dataset(r'../regression/regression.train')
lgb.train({'objective': 'regression', 'weight_column': 5, 'group_column': 7}, train)
```

What note should be written for [`save_binary`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#save_binary) and [`enable_load_from_binary_file`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#enable_load_from_binary_file)?

### CODE REVIEW

**Comment by StrikerRUS** [comment]
Usage example:

```
train = lgb.Dataset(r'../regression/regression.train')
lgb.train({'objective': 'regression', 'weight_column': 5, 'group_column': 7}, train)
```

**Comment by StrikerRUS** [comment]
Hmmm, according to the log
```
[LightGBM] [Info] Using column number 5 as weight
[LightGBM] [Info] Using column number 7 as group/query id
---->  [LightGBM] [Info] Loading initial scores...  <----
[LightGBM] [Info] Loading query weights...
[LightGBM] [Info] Total Bins 5632
[LightGBM] [Info] Number of data: 7000, number of used features: 26
```
it seems to me that the **Note** for [`initscore_filename`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#initscore_filename) and [`valid_data_initscores`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#valid_data_initscores) should be fixed from `Note: can be used only in CLI version` to `Note: works only in case of loading data directly from file`. Am I right?

**Comment by guolinke** [comment]
Yeah you are right

**Comment by StrikerRUS** [comment]
@guolinke Thanks for the clarification!

And what about `*binary*` params?
> What note should be written for [`save_binary`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#save_binary) and [`enable_load_from_binary_file`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#enable_load_from_binary_file)?

**Comment by guolinke** [comment]
I think they are `Note: works only in case of loading data directly from file`

**Comment by StrikerRUS** [comment]
Shouldn't the note for `save_binary` be `Note: can be used only in CLI version; for language-specific packages you can use the correspondent function`?

https://github.com/Microsoft/LightGBM/blob/90127b526be8b13bdf840effacde9e61fa765a6f/src/application/application.cpp#L113-L116
https://github.com/Microsoft/LightGBM/blob/462612b42c6fe7bb1be1ccd8b72d77021b9f2d38/src/c_api.cpp#L808-L814
https://github.com/Microsoft/LightGBM/blob/b37065dbd5510079aac341a34895f4854c782fcb/R-package/R/lgb.Dataset.R#L567-L577

And `enable_load_from_binary_file` seems to be unused at all: https://github.com/Microsoft/LightGBM/search?q=enable_load_from_binary_file&unscoped_q=enable_load_from_binary_file

What am I missing?..

**Comment by guolinke** [comment]
sorry, you are right about save_binary.
as for enable_load_from_binary_file, I think it is a deprecated parameter.

**Comment by StrikerRUS** [comment]
Thank you for the confirmation!

> as for enable_load_from_binary_file, I think it is a deprecated parameter.

Should it be removed then?

**Comment by guolinke** [comment]
@StrikerRUS 
Yeah, they can be removed.

**Comment by StrikerRUS** [comment]
Done!
